### PR TITLE
Add custom_image_name attribute to rule docker_build.

### PR DESCRIPTION
**Pull request by linuxerwang**

### CODE REVIEW

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

**Comment by linuxerwang** [comment]
Please see issue #616.

File: tools/build_defs/docker/build.bzl [hunk 1]
```diff
@@ -219,12 +219,15 @@ def _compute_layer_name(ctx, layer):
 
 def _repository_name(ctx):
   """Compute the repository name for the current rule."""
+  if ctx.attr.image_name:
+    return os.path.join(ctx.attr.repository, ctx.label.image_name)
+
   if ctx.attr.legacy_repository_naming:
     # Legacy behavior, off by default.
-    return "%s/%s" % (ctx.attr.repository, ctx.label.package.replace("/", "_"))
+    return os.path.join(ctx.attr.repository, ctx.label.package.replace("/", "_"))
```
Comment (mattmoor) [inline tools/build_defs/docker/build.bzl]: s/os.path.join/_join_path/

I meant the equivalent of this std Python library, which for Bazel is defined in `path.bzl`

The same below.

File: tools/build_defs/docker/build.bzl [hunk 2]
```diff
@@ -219,12 +219,15 @@ def _compute_layer_name(ctx, layer):
 
 def _repository_name(ctx):
   """Compute the repository name for the current rule."""
+  if ctx.attr.image_name:
```
Comment (mattmoor) [inline tools/build_defs/docker/build.bzl]: Path joining should replace this.

File: tools/build_defs/docker/build.bzl (hunk 1 above)
Comment (linuxerwang) [inline tools/build_defs/docker/build.bzl]: Done.

File: tools/build_defs/docker/build.bzl (hunk 2 above)
Comment (linuxerwang) [inline tools/build_defs/docker/build.bzl]: Done.

**Comment by damienmg** [comment]
FTR: the change is merge internally but we have some problem with our tooling atm so it might not appear before monday

**Comment by linuxerwang** [comment]
Thanks, @damienmg!

**Pull request closed without merging**